android/app/src/main/java/org/electroncash/electroncash3/NewWallet.kt

```diff
@@ -225,32 +231,35 @@ abstract class NewWalletDialog2 : TaskLauncherDialog<String>() {
         return name
     }
 
-    abstract fun onCreateWallet(name: String, password: String)
+    abstract fun onCreateWallet(name: String, password: String): PyObject?
 
     override fun onPostExecute(result: String) {
-        val multisig = arguments!!.getBoolean("multisig")
+        val keystores = arguments!!.getStringArrayList("keystores")
 
         /**
          * For multisig wallets, we need to first show the master key to the 1st cosigner, and
          * then prompt for data for all other cosigners by calling the KeystoreDialog again.
          */
-        if (multisig) {
-            val currentCosigner = arguments!!.getInt("i_signer")
+        if (keystores != null) {
+            val currentCosigner = keystores.size
             val numCosigners = arguments!!.getInt("cosigners")
 
             if (currentCosigner < numCosigners) {
                 // The first cosigner sees their master public key; others are prompted for data
-                val nextDialog: DialogFragment = if (currentCosigner == 1) {
-                    MasterPublicKeyDialog()
-                } else {
-                    KeystoreDialog()
+                if (currentCosigner == 1) {
+                    (targetFragment as DialogFragment).dismiss()
+                    val nextDialog: DialogFragment = MasterPublicKeyDialog()
+                    nextDialog.setArguments(arguments)
```

Comment:
The previous keystore bugs have all been fixed, but there are some new ones caused by sharing arguments objects between dialogs:

* Set cosigners to 3, and enter the first two keys or seeds. Then press Back. The Master public key dialog appears unchanged, but because all the dialogs have been sharing the same arguments object, this dialog's arguments now has 2 keystores. So when you rotate the screen, the key will change from the first to the second one. And when you press Next, instead of letting you re-enter cosigner 2, it goes straight to cosigner 3.

* Ensure no wallet is open (otherwise Main.kt would discard saved state after a process restart). Set cosigners to 3, and enter the first key or seed. Then when the NewWalletDialog2 appears for the second key or seed, press Home to send the app to the background, and kill the process from Android Studio. When you re-launch the app, the dialogs will reappear as they were before. And when you enter the second key or seed, the KeystoreDialog appears for cosigner 3. However, because its arguments were serialized when the app went into the background, then deserialized back into the new process, they are now NOT sharing the same arguments object as the NewWalletDialog2. So when NewWalletDialog2 added the second keystore to its own arguments, this had no effect on the KeystoreDialog arguments. And sure enough, when you rotate the screen, the KeystoreDialog regenerates its title and the cosigner number changes from 3 to 2.

I think the simplest solution is to copy the arguments for each new dialog rather than sharing them. So everywhere that currently calls `setArguments(arguments)` should instead be operating on a copy created by calling `Bundle(arguments)`. And a `setArguments` call should also be made when NewWalletDialog2 updates the existing KeystoreDialog. In fact, it would be better to combine the two places that set the KeystoreDialog title, by moving that code into a KeystoreDialog method, and calling it from NewWalletDialog2 after updating the KeystoreDialog arguments.

For the same reason, updateKeystores should create a new ArrayList each time rather than appending to the existing one.